### DISCUSSION TITLE
Why params[..] values are not set when paging or or time-range is not supported

### DISCUSSION THREAD

**Discussion by return42** [searxng/searxng#1559] [Q&A] [answered]
May be my code analyses is wrong or I don't understand why and when `params[..]` needs to be set in the search processor.

@dalf: may be you can point out where I fail .. (the implementations are from you)

In the abstract class by example:

https://github.com/searxng/searxng/blob/50ec821e8c5b099947d657722b1879fc8c03bcfa/searx/search/processors/abstract.py#L140-L160

Arguments like `language` or `safesearch` are not passed to the `params[..]` when paging or time-range is not supported by the engine.

And in the online processor ...

https://github.com/searxng/searxng/blob/50ec821e8c5b099947d657722b1879fc8c03bcfa/searx/search/processors/online.py#L52-L61

the default params and the User-Agent header are not set when the abstract class (called by `super`) returns `None`.

This behavior was introduced in c4067a9d02d97c in the [`def _is_accepted(...)`](https://github.com/searxng/searxng/commit/c4067a9d02d97ca08259daeede92a031cd608724#diff-f5962794ee51e4631700f577aa174f19623c87377130020dba94bd55410f5420R447-R464) function.

Before this commit the (online) `params[..]` has been set whenever the [engine was not an offline engine](https://github.com/searxng/searxng/commit/c4067a9d02d97ca08259daeede92a031cd608724#diff-f5962794ee51e4631700f577aa174f19623c87377130020dba94bd55410f5420L482-L495)

**Comment by dalf** [comment, marked as answer]
> Arguments like language or safesearch are not passed to the params[..] when paging or time-range is not supported by the engine.

If you are talking about this condition:

```python
if search_query.pageno > 1 and not self.engine.paging: 
    return None
```

-->

If user query is not asking for the page one = page two, three, ... (1) and the engine doesn't support paging (2) then the engine won't get the query (3).

* (1) `search_query.pageno > 1`
* (2) `not self.engine.paging`
* (3) `return None` : in the following code fragment, `continue` means the query is skipped for this engine: https://github.com/searxng/searxng/blob/50ec821e8c5b099947d657722b1879fc8c03bcfa/searx/search/__init__.py#L95-L98

---

Same for `time_range_support` : if a specific date range is asked and the engine does not support time range, then the engine don't get the query.

**Reply by return42** [reply]
Ah! .. thanks a lot .. what I missed was:

> * (3) `return None` : in the following code fragment, `continue` means the query is skipped for this engine: 

https://github.com/searxng/searxng/blob/50ec821e8c5b099947d657722b1879fc8c03bcfa/searx/search/__init__.py#L95-L98
